File: android/ValleCOM/app/src/main/java/com/valleapp/valleCOM/Activitys/Refill.java
package com.valleapp.valleCOM.Activitys;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.ListView;

import com.valleapp.valleCOM.R;
import com.valleapp.valleCOM.adaptadores.AdaptadorRefill;
import com.valleapp.valleCOM.db.DBCuenta;
import com.valleapp.valleCOM.utilidades.ActivityBase;

import org.json.JSONObject;

import java.util.ArrayList;

public class Refill extends ActivityBase {


    DBCuenta dbCuenta = new DBCuenta(this);
    String idMesa;

    public void mostrarLista(){
        ListView ls = findViewById(R.id.listaPedidosRefill);
        ArrayList<JSONObject> lista = dbCuenta.getPedidosChoices(idMesa);
        ls.setAdapter(new AdaptadorRefill(this, lista));
        ls.setOnItemClickListener((adapterView, view, i, l) -> {
            try{
            Intent it = getIntent();
            it.putExtra("IDPedido", lista.get(i).getString("IDPedido"));
            setResult(RESULT_OK, it);
            finish();
            }catch (Exception e){
                e.printStackTrace();
            }
        });
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_refill);
        idMesa = getIntent().getExtras().getString("id_mesa");
    }

    @Override
    protected void onResume() {
        super.onResume();
        mostrarLista();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: android/ValleTPV/ValleCOM/src/main/java/com/valleapp/vallecom/activitys/Refill.kt
package com.valleapp.vallecom.activitys


import android.os.Bundle
import android.util.Log
import android.widget.ListView
import com.valleapp.vallecom.adaptadores.AdaptadorRefill
import com.valleapp.vallecom.utilidades.ActivityBase
import com.valleapp.valletpv.R
import com.valleapp.valletpvlib.db.DBCuenta

class Refill : ActivityBase() {

    private val dbCuenta = DBCuenta(this)
    private lateinit var idMesa: String

    private fun mostrarLista() {
        val ls = findViewById<ListView>(R.id.listaPedidosRefill)
        val lista = dbCuenta.getPedidosChoices(idMesa)
        val adaptador = AdaptadorRefill(this, lista)
        ls.adapter = adaptador.toListAdapter() // Convierte AdaptadorRefill a un adaptador compatible con ListView
        ls.setOnItemClickListener { _, _, i, _ ->
            try {
                val it = intent
                it.putExtra("IDPedido", lista[i].getString("IDPedido"))
                setResult(RESULT_OK, it)
                finish()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_refill)
        idMesa = intent.extras?.getString("id_mesa") ?: ""
        Log.d("Refill", "onCreate: idMesa = $idMesa")
    }

    override fun onResume() {
        super.onResume()
        mostrarLista()
        Log.d("Refill", "onResume: Lista mostrada")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("Refill", "onDestroy: Activity destruida")
    }
}

